base/sort.jl

```diff
@@ -426,6 +426,23 @@ struct InsertionSortAlg <: Algorithm end
 struct QuickSortAlg     <: Algorithm end
 struct MergeSortAlg     <: Algorithm end
 
+"""
+    AdaptiveSort(fallback)
+
+Indicate that a sorting function should pick the fastest available algorithm for the
+given element type, order, length, and contents. If no alternatives are viable, uses
+the `fallback` algorithm. AdaptiveSort is stable if `fallback` is stable.
+
+Algorithms currently in use:
+  * Insertion sort for short lists
+  * Radix sort for long lists
+  * Counting sort for lists of integers spanning a short range
+  * Fallback algorithm when the input cannot be efficiently
+    converted into a list of integers maintaining sort order
```

Comment:
I don't understand this. Does this refer to the `fallback`? Also "list" isn't typical Julia terminology, better say "vector" (here and below).

Reply:
I've rewritten it. Does the docstring make sense now?